tools/scripts/uconfig_vars_test.py

```diff
@@ -0,0 +1,217 @@
+    # execute ICU4C unit tests or/and header test.
+    for uconfig_no in all_uconfig_no_flags:
+        with open('common/unicode/uconfig.h', 'w') as file_handle:
+            file_handle.write(EnableUConfigNo(orig_uconfig_file, uconfig_no))
+        RunCmd('make clean')
+
+        # Run ICU4C unit tests if requested, except for the excluded flags.
+        if run_unit and uconfig_no not in excluded_unit_test_flags:
+            print('Running unit tests with %s set to 1.' % uconfig_no)
+            _, exit_code = RunCmd('make -j2 check')
+            if exit_code == 0:
+                test_results[uconfig_no]['unit_test'] = True
+            else:
+                test_results[uconfig_no]['unit_test'] = False
+
+        # If requested, run header tests.
+        if run_hdr:
+            print('Running header tests with %s set to 1.' % uconfig_no)
+            _, exit_code = RunCmd(
+                'PATH=/tmp/icu_cnfg/bin:$PATH make -C test/hdrtst check')
+            if exit_code == 0:
+                test_results[uconfig_no]['hdr_test'] = True
+            else:
+                test_results[uconfig_no]['hdr_test'] = False
+
+    # If unit test run is requested, run unit tests with all flags set to '1'
+    # except for excluded flags.
+    if run_unit:
+        uconfig_no_all = orig_uconfig_file
+        for uconfig_no in all_uconfig_no_flags:
+            if uconfig_no not in excluded_unit_test_flags:
+                uconfig_no_all = EnableUConfigNo(uconfig_no_all, uconfig_no)
+        with open('common/unicode/uconfig.h', 'w') as file_handle:
+            file_handle.write(uconfig_no_all)
+        RunCmd('make clean')
+
+        print('Running unit tests with all flags set to 1.')
+        _, exit_code = RunCmd('make -j2 check')
+        if exit_code == 0:
+            test_results['all_flags']['unit_test'] = True
+        else:
+            test_results['all_flags']['unit_test'] = False
+
+    # If header test run is requested, run the tests with all flags set to '1'.
+    if run_hdr:
+        uconfig_no_all = orig_uconfig_file
+        for uconfig_no in all_uconfig_no_flags:
+            uconfig_no_all = EnableUConfigNo(uconfig_no_all, uconfig_no)
+        with open('common/unicode/uconfig.h', 'w') as file_handle:
+            file_handle.write(uconfig_no_all)
+        print('Running header tests with all flags set to 1.')
+        hdr_test_result_all, exit_code = RunCmd(
+            'PATH=/tmp/icu_cnfg/bin:$PATH make -C test/hdrtst check')
+        if exit_code == 0:
+            test_results['all_flags']['hdr_test'] = True
+        else:
+            test_results['all_flags']['hdr_test'] = False
+
+    # Review test results and report any failures.
+    outcome = 0
+    print('Summary:\n')
+    for uconfig_no in all_uconfig_no_flags:
+        if run_unit and uconfig_no not in excluded_unit_test_flags:
+            if test_results[uconfig_no]['unit_test'] == False:
```

Comment:
Please change all `condition == False` to `not condition`.
And please add parentheses around that if it's after an `other_condition and ...`

Reply:
Done throughout.